ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceListener.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@class CKTransactionalComponentDataSource;
+@class CKTransactionalComponentDataSourceAppliedChanges;
+@class CKTransactionalComponentDataSourceState;
+
+@protocol CKTransactionalComponentDataSourceListener
+
+/**
+ Announced on the main thread when the data source has just updated its state.
+ @param dataSource The sending data source; its state property now returns an updated object.
```

Comment:
Hmm, so from a client perspective what's the contract on when this is called?  Why not vend the new state object here?